modules/auto-update/implementation/ot-auto-updater.js

```diff
@@ -0,0 +1,227 @@
+            if (this.config.executeOnComplete) await this.promiseBlindExecute(this.config.executeOnComplete);
+            process.exit(1);
+        }catch(e) {
+            this.logger.error(`AutoUpdater - Error updating application. Error message: ${e.message}`);
+            return false;
+        }
+    }
+
+    /**
+     * Copy the files to the app directory, and install new modules
+     * The update is installed from  the configured tempLocation.
+     */
+    async installUpdate() {
+        const source = path.join(this.config.tempLocation, CLONE_SUBDIRECTORY);
+        const destination = appRootPath.path;
+        this.logger.info('AutoUpdater - Installing update...');
+        this.logger.info(`AutoUpdater - Source: ${source}`);
+        this.logger.info(`AutoUpdater - Destination: ${destination}`);
+        await fs.ensureDir(destination);
+        await fs.copy(source, destination);
+    }
+
+    /**
+     * Reads the applications version from the package.json file.
+     */
+    readAppVersion() {
+        const file = path.join(appRootPath.path, 'package.json');
+        this.logger.info(`AutoUpdater - Reading app version from ${file}`);
+        const appPackage = fs.readFileSync(file);
+        return JSON.parse(appPackage).version;
+    }
+
+    /**
+     * A promise wrapper for sending a get https requests.
+     * @param {String} url - The Https address to request.
+     * @param {String} options - The request options. 
+     */
+    promiseHttpsRequest(url, options) {
+        return new Promise((resolve, reject) => {
+            const req = https.request(url, options, res => {
+                let body = '';
+                res.on('data', data => {body += data});
+                res.on('end', () => {
+                    if (res.statusCode === 200) return resolve(body);
+                    this.logger.info(`AutoUpdater - Bad Response ${res.statusCode}`);
+                    reject(res.statusCode);
+                });
+            });
+            this.logger.info(`AutoUpdater - Sending request to ${url}`);
+            this.logger.info(`AutoUpdater - Options: ${JSON.stringify(options)}`);
+            req.on('error', reject);
+            req.end();
+        }); 
+    }
+
+    /**
+     * Reads the applications version from the git repository.
+     */
+    async readRemoteVersion() {
+        const options = {}
+        let url =  `${REPOSITORY_URL}/${this.config.branch}/package.json`;
+        if (url.includes('github')) url = url.replace('github.com', 'raw.githubusercontent.com');
+        this.logger.info(`AutoUpdater - Reading remote version from ${url}`);
+        
+        try {
+            const body = await this.promiseHttpsRequest(url, options);
+            const remotePackage = JSON.parse(body);
+            const {version} = remotePackage;
+            return version;
+        } catch(e) {
+            this.logger.error(`This repository requires a token or does not exist. Error message: ${e.message}`);
```

Comment:
We should probably throw an error here if we are not able to read remote version